### PR TITLE
Add javadoc and update DG

### LINKED ISSUE #102: Add JavaDoc comments
* budgetcommand/budget
* ui
* storage
* overview
* balance

### PR DESCRIPTION
**Pull request by ryan1604**
Fixes #102

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+    * [Delete budget](#delete-budget)
+    * [Reset budget](#reset-budget)
+    * [View budget](#view-budget)
+  * [Product Scope](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+  * [User Stories](#user-stories)
+  * [Non-Functional Requirements](#non-functional-requirements)
+  * [Glossary](#glossary)
+  * [Instructions for manual testing](#instructions-for-manual-testing)
```
Comment (NeoMinWei) [inline docs/DeveloperGuide.md:29]: nice addition

File: src/main/java/seedu/financialplanner/commands/BudgetCommand.java [hunk 2]
```diff
@@ -33,6 +42,12 @@ public BudgetCommand(RawCommand rawCommand) throws FinancialPlannerException {
         }
     }
 
+    /**
+     * Checks if budget is a positive number and is lower than or equal to total balance.
+     *
+     * @param rawCommand The raw command containing arguments and parameters given by the user.
+     * @throws FinancialPlannerException If there is an issue with budget format.
+     */
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/BudgetCommand.java:50]: need javadoc for private methods also?

**Review by NeoMinWei — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/commands/BudgetCommand.java (hunk 2 above)
Comment (ryan1604) [reply src/main/java/seedu/financialplanner/commands/BudgetCommand.java:50]: no

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (wwweert123) [reply docs/DeveloperGuide.md:29]: 👍

File: docs/diagrams/Budget.puml [hunk 3]
```diff
@@ -2,10 +2,9 @@
 
 participant ":BudgetCommand" as BudgetCommand
 participant "<<class>>\nBudget" as Budget
-participant "<<class>>\nUi" as Ui
+participant ":Ui" as Ui
```
Comment (wwweert123) [inline docs/diagrams/Budget.puml:5]: Nice

File: src/main/java/seedu/financialplanner/FinancialPlanner.java [hunk 4]
```diff
@@ -10,21 +11,35 @@
 
 import java.time.LocalDate;
 
+/**
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/FinancialPlanner.java:14]: Very good comments

File: src/main/java/seedu/financialplanner/cashflow/Budget.java [hunk 5]
```diff
@@ -32,50 +51,104 @@ public static void updateBudget(double budget) {
         assert initialBudget == budget : "Initial budget should be equal to updated budget";
     }
 
+    /**
+     * Returns the current budget.
+     *
+     * @return The current budget.
+     */
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/cashflow/Budget.java:58]: I dont think javadoc  for getters and setters are necessary  :(((((((((((((((((((((((

Comment (wwweert123) [inline src/main/java/seedu/financialplanner/cashflow/Budget.java:58]: I dont think javadocs for getters and setters are necessary :(((

File: src/main/java/seedu/financialplanner/FinancialPlanner.java (hunk 4 above)
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/FinancialPlanner.java:14]: Very good comments 💯

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (wwweert123) [reply docs/DeveloperGuide.md:29]: 👍

**Review by wwweert123 — APPROVED** [review]
LGTM

**Pull request merged**